packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js

```diff
@@ -19,7 +19,13 @@ async function run(context, distributionDirPath) {
 
   const hasCloudFront = !!context?.exeInfo?.template?.Resources?.CloudFrontDistribution;
 
-  fileList.forEach(filePath => {
+
+  await Promise.all(fileList.map(async filePath => {
+    if(!filePath.includes('index.html')) {
+      return await uploadFileTasks.push(() => uploadFile(s3Client, hostingBucketName, distributionDirPath, filePath, hasCloudFront));
+  }}));
+
+  fileList.filter(filePath => filePath.includes('index.html')).forEach(filePath => {
     uploadFileTasks.push(() => uploadFile(s3Client, hostingBucketName, distributionDirPath, filePath, hasCloudFront));
```

Comment:
I suggest to refactor this into two sections for better readability. something like
```
const indexFile = ... select index from fileList...;
const otherFiles = ... select all other files from fileList....;

await Promise.all(...upload all files...);
await upload(index);

```

Reply:
Done 3abfb5164c510f1697bc828246313a66f817646e